scripts/travis-setup.sh

```diff
@@ -1,25 +1,37 @@
 #!/bin/bash
-set -e
+set -euo pipefail
 
 SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 
 if [[ $TRAVIS_OS_NAME = "linux" ]]
 then
-    docker pull bioconda/bioconda-builder
+    tag=Linux
 else
+    tag=MacOSX
+fi
+echo $(which python)
+curl -O https://repo.continuum.io/miniconda/Miniconda3-latest-${tag}-x86_64.sh
+sudo bash Miniconda3-latest-${tag}-x86_64.sh -b -p /anaconda
+sudo chown -R $USER /anaconda
+export PATH=/anaconda/bin:$PATH
+conda install -y --file scripts/requirements.txt
+
+# Install bioconda-utils with miniconda version of python
+git clone https://github.com/bioconda/bioconda-utils.git
+(cd bioconda-utils && python setup.py install)
+
 
-    # install conda
-    curl -O https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
-    sudo bash Miniconda3-latest-MacOSX-x86_64.sh -b -p /anaconda
-    sudo chown -R $USER /anaconda
+if [[ $TRAVIS_OS_NAME != "linux" ]]
```

Comment:
Currently, yes: [this line](https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/utils.py#L281) sends the channels to the docker container but if not `docker` then [this line](https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/utils.py#L303) does not.

I think a better solution might be to add the channels to the config.yml; then the only special-casing will be whether to use `--docker` or not.